contracts/NestedAsset.sol

```diff
@@ -84,54 +102,38 @@ contract NestedAsset is ERC721Enumerable, OwnableFactoryHandler {
         return tokenId;
     }
 
-    /// @notice Mints an ERC721 token and sets the tokenUri.
-    /// Only the NestedFactory contract can call this function.
-    /// @param _owner The account address that signed the transaction
-    /// @param _metadataURI he metadata URI string
-    /// @param _replicatedTokenId The token id of the replicated asset, 0 if no replication
-    /// @return The minted token's id
-    function mintWithMetadata(
-        address _owner,
-        string memory _metadataURI,
-        uint256 _replicatedTokenId
-    ) external returns (uint256) {
-        uint256 tokenId = mint(_owner, _replicatedTokenId);
-        _setTokenURI(tokenId, _metadataURI);
-        return tokenId;
-    }
-
-    /// @notice Backfills the token URI if it had never set
-    /// @param _tokenId The id of the NestedAsset
-    /// @param _owner The id of the NestedAsset
-    /// @param _metadataURI The metadata URI string
-    function backfillTokenURI(
-        uint256 _tokenId,
-        address _owner,
-        string memory _metadataURI
-    ) external onlyFactory onlyTokenOwner(_owner, _tokenId) {
-        require(bytes(tokenURI(_tokenId)).length == 0, "NA: TOKEN_URI_IMMUTABLE");
-        _setTokenURI(_tokenId, _metadataURI);
-    }
-
     /// @notice Burns an ERC721 token
     /// @param _owner The account address that signed the transaction
     /// @param _tokenId The id of the NestedAsset
     function burn(address _owner, uint256 _tokenId) external onlyFactory onlyTokenOwner(_owner, _tokenId) {
         lastOwnerBeforeBurn[_tokenId] = _owner;
         _burn(_tokenId);
+    }
 
-        if (bytes(_tokenURIs[_tokenId]).length != 0) {
-            delete _tokenURIs[_tokenId];
-        }
+    /* ----------------------------- ONLY OWNER ---------------------------- */
+
+    /// @notice Reveal the token URI (with API)
+    function reveal() external onlyOwner {
+        isRevealed = true;
```

Comment:
There is no mechanism to unreveal to correct a mistake. Consider changing this function to `setIsRevealed(bool _isRevealed)`.

Reply:
👍